pyglider/slocum.py

```diff
@@ -848,29 +878,58 @@ def binary_to_timeseries(indir, cachedir, outdir, deploymentyaml, *,
     for atts in ncvar[name].keys():
         if (atts != 'coordinates') & (atts != 'units') & (atts != 'calendar'):
             attr[atts] = ncvar[name][atts]
-    sensors = [time_base]
 
+    sensors = []
     for nn, name in enumerate(thenames):
         sensorname = ncvar[name]['source']
         if not sensorname == time_base:
             sensors.append(sensorname)
         else:
             baseind = nn
-
-    print(*sensors)
-    # get the data, with `time_base` as the time source that
-    # all other variables are synced to:
-    data = list(dbd.get_sync(*sensors))
-    # get the time:
-    time = data.pop(0)
+            sensors.insert(0, sensorname)
+    _log.debug(f'sensors: {[i for i in sensors]}')
+
+    time_base_union = time_base == 'union'
+    if time_base_union:  
```

Comment:
Please put the "normal" loop first, and the time_base_union loop second.  It makes the code easier to read if the default predeces the non-default.